Fix entity tag picker (#15217)
backstage/backstage: .changeset/funny-pianos-grow.md

---
'@backstage/plugin-catalog-react': patch
---

> Fix filter bug that leaves filters selected when they are not available for the selected kind.

Reviewer: I hope I did not miss anything in this sentence, the important part was to highlight the specific picker.

```suggestion
Fixed bug in `EntityTagPicker` that filtered on unavailable tags for the selected kind.
```

Author: That makes sense! Is it possible to also mention the other pickers that were updated in this PR https://github.com/backstage/backstage/pull/15058 since it did not have a changeset when it was merged?

---
'@backstage/plugin-catalog-react': patch
---

> Fixed bug in `EntityTagPicker` that filtered on unavailable tags for the selected kind.

Reviewer: From my understanding we only update `EntityTagPicker` or did I miss something? I think we should remove the other lines below if that's the case :)

Author: I updated the other pickers in a different PR that was merged before I could add a changeset. I can totally remove these lines I was just hoping we could add them in since they were not added before. This is the other pr https://github.com/backstage/backstage/pull/15058

Reviewer: Oh it looks like that indeed snuck in without a changeset, thanks for pointing it out @sarahpmedeiros! 

It has however already been released, in that case could you instead update the 1.2.2 section in https://github.com/backstage/backstage/blob/master/plugins/catalog-react/CHANGELOG.md#122 and add a note about the previous fixes in the list of patch changes.

And then just have this changeset only mention `EntityTagPicker`

Author: will do, updated this changeset thank you!

Author: https://github.com/backstage/backstage/pull/15377 also created the other pr
